Training/configs/training_v1.yaml

```diff
@@ -38,251 +35,374 @@ CellObjectType : [
+                - tau_pt_weighted_dr_iso: [null, null, "linear", 0., 1.]
+                - tau_leadingTrackNormChi2: [null, null, "normal", -5, 5]
+                - tau_e_ratio_valid: [null, {"tau_e_ratio_valid": *tau_e_ratio_valid}, "no_scaling"]
+                - tau_e_ratio: [*tau_e_ratio_valid, null, "linear", 0., 1.]
+                - tau_gj_angle_diff_valid: [null, {"tau_gj_angle_diff_valid": *tau_gj_angle_diff_valid}, "no_scaling"]
+                - tau_gj_angle_diff: [*tau_gj_angle_diff_valid, null, "linear", 0., *pi]
+                - tau_n_photons: [null, null, "normal", -5, 5]
+                - tau_emFraction: [null, null, "linear", -1., 1.]
+                - tau_inside_ecal_crack: [null, null, "no_scaling"]
+                - tau_leadChargedCand_etaAtEcalEntrance_minus_tau_eta: [null, {"tau_leadChargedCand_etaAtEcalEntrance_minus_tau_eta": "tau_leadChargedCand_etaAtEcalEntrance - tau_eta"}, "normal", -5, 5]
 
-    PfCand_electron : [
-                "pfCand_ele_valid",
-                "pfCand_ele_rel_pt",
-                "pfCand_ele_deta",
-                "pfCand_ele_dphi",
-                "pfCand_ele_pvAssociationQuality",
-                "pfCand_ele_puppiWeight",
-                "pfCand_ele_charge",
-                "pfCand_ele_lostInnerHits",
-                "pfCand_ele_nPixelHits",
-                "pfCand_ele_vertex_dx",
-                "pfCand_ele_vertex_dy",
-                "pfCand_ele_vertex_dz",
-                "pfCand_ele_vertex_dx_tauFL",
-                "pfCand_ele_vertex_dy_tauFL",
-                "pfCand_ele_vertex_dz_tauFL",
-                "pfCand_ele_hasTrackDetails",
-                "pfCand_ele_dxy",
-                "pfCand_ele_dxy_sig",
-                "pfCand_ele_dz",
-                "pfCand_ele_dz_sig",
-                "pfCand_ele_track_chi2_ndof",
-                "pfCand_ele_track_ndof"
-                ]
+    PfCand_electron :
+                - pfCand_ele_valid: [*is_pfElectron, null, "no_scaling"]
+                - pfCand_ele_rel_pt: [*is_pfElectron, {"pfCand_ele_rel_pt": "pfCand_pt/tau_pt"}, "normal", -5, 5]
+                - pfCand_ele_deta: [*is_pfElectron, {"pfCand_ele_deta": "pfCand_eta - tau_eta"}, "linear", {'inner': [-0.1, 0.1], 'outer': [-0.5, 0.5]}]
+                - pfCand_ele_dphi: [*is_pfElectron, {"pfCand_ele_dphi": "pfCand_phi - tau_eta"}, "linear", {'inner': [-0.1, 0.1], 'outer': [-0.5, 0.5]}] # NB: will not bring dphi values to [-pi,pi] on the fly here
+                - pfCand_ele_pvAssociationQuality: [*is_pfElectron, {"pfCand_ele_pvAssociationQuality": "pfCand_pvAssociationQuality"}, "categorical"]
+                - pfCand_ele_puppiWeight: [*is_pfElectron, {"pfCand_ele_puppiWeight": "pfCand_puppiWeight"}, "no_scaling"]
+                - pfCand_ele_charge: [*is_pfElectron, {"pfCand_ele_charge": "pfCand_charge"}, "no_scaling"]
+                - pfCand_ele_lostInnerHits: [*is_pfElectron, {"pfCand_ele_lostInnerHits": "pfCand_lostInnerHits"}, "categorical"]
```

Comment:
@yaourtpourtoi @kandrosov if I did not mix it up, I have a question about the following variables: `pfCand_pvAssociationQuality`, `pfCand_lostInnerHits` and also to the `pfCand_fromPV` which we use for the muon collection: did we agree that all variables should be scaled to [-1, +1]? In this I case, we need to use linear mapping on this interval (because these variables are not distributed within [-1, +1]), yes it is categorical, but we are not handling it appropriately on the NN input (not embedding them or something), so wouldn't this feature just negatively influence the scale inside NN?

For instance looking at the `pfCand_pvAssociationQuality`:
![image](https://user-images.githubusercontent.com/31743376/130472245-1c2efd11-17d6-4194-969d-1a749766ed42.png)

Reply:
Yes, these variables as of now definitely go beyond the [-1, 1] scale. However, it's hard for me personally to predict intuitively whether this can cause troubles for the training/overall performance. For example, `pfCand_pvAssociationQuality` seems to have mostly values 8, so I can speculate that this feature might be considered as trivially redundant by the model (and therefore the different scale doesn't cause problems). So I'd personally ran experiments to see if the scaling of those features changes something in terms of performance/convergence (or maybe even if the scaling is needed at all..). And additionally, it would be interesting to visualise the input tensor (like a 3D heatmap) as it goes through the first layers of the model to see, how the inputs with "out-of-scale" values propagate through the network - e.g. if the amplitude is getting magnified or on the contrary diluted back to average scale.